core/vm/gas.go

```diff
@@ -32,21 +32,16 @@ const (
 
 // callGas returns the actual gas cost of the call.
 //
-// The cost of gas was changed during the homestead price change HF.
-// As part of EIP 150 (TangerineWhistle), the returned gas is gas - base * 63 / 64.
-func callGas(isEip150 bool, availableGas, base uint64, callCost *uint256.Int) (uint64, error) {
-	if isEip150 {
-		availableGas = availableGas - base
-		gas := availableGas - availableGas/64
-		// If the bit length exceeds 64 bit we know that the newly calculated "gas" for EIP150
-		// is smaller than the requested amount. Therefore we return the new gas instead
-		// of returning an error.
-		if !callCost.IsUint64() || gas < callCost.Uint64() {
-			return gas, nil
-		}
-	}
-	if !callCost.IsUint64() {
-		return 0, ErrGasUintOverflow
```

Comment:
why don't we need to say that the gas in overflowed?.

Reply:
In the EIP150 case, an overflow returns 'gas' which is what I have replicated here.